### DISCUSSION TITLE
Network analysis and headings support

### DISCUSSION THREAD

**Discussion by quicly** [chrisgrieser/obsidian-quadro#4] [Ideas]
First of all, thank you for a great plugin. I have already used Obsidian for qualitative analysis, but before, I mostly did coding and reading in other software and manually transferred files into Obsidian for more advanced tasks.

Moreover, the idea of this plugin works not only for qualitative data analysis but also for a PKM style that is closer to me, in which notes stay connected with their source text but can be classified and categorized separately (this is the reason why I still used Citavi and not Zotero/Obsidian for my academic work). I have always wanted to move my workflow fully to Obsidian. The way you have realized code support is similar to my desired workflow.

With that in mind, here are a couple of suggestions and questions:

1) I think it would be good to add a network analysis approach. There is a plugin called Graph Analysis, which I use extensively in my qualitative work. Although it hasn't been updated for 2 years, you can take some ideas from it. For example, analysis of similarity between data files depending on common codes used in them; and similarity of the codes depending on co-occurrences of other codes (for example, if code1 often co-occurs with code2 and code3 often co-occurs with code2, then code 1 and code 3 have some degree of similarity).

2) Right now, the coding does not support headings. Although I understand why that is so, there are cases when data have a structure like this:

Heading
content

When assigning the code to the content, you lose the heading in a separate code note. So, if there is a way to keep headings of the paragraph while assigning code, that would be great. Maybe there is a way to assign a code to the heading itself, which would then include the content that follows it?

3) Sometimes I use the technique of comprehensive coding, which means coding every paragraph in the text. In such cases, it would be useful to implement a function to search for any paragraphs that haven't been coded.

**Comment by chrisgrieser** [comment]
Thank you! Yeah, "unifying" everything in Obsidian for ease of use was exactly the idea behind Quadro :)

> I think it would be good to add a network analysis approach. There is a plugin called Graph Analysis, which I use extensively in my qualitative work. Although it hasn't been updated for 2 years, you can take some ideas from it.

I agree, however, creating any sort of network analysis is going to be quite a lot of effort and thus somewhat out of the scope for this project. The idea behind Quadro is more or less to create the basics needed for qualitative research and leaving specialized stuff like this more general Obsidian plugins (since I as one person will never be able to create all the features of a company-backed app like MaxQDA). I think it'll be more fruitful for someone to continue working on Graph Analysis this into Quadro. 

> Right now, the coding does not support headings. Although I understand why that is so, there are cases when data have a structure like this:

As far as I can tell, that is unfortunately a restriction of how block-links in Obsidian work, and not something I can change on my end, sorry.

> Sometimes I use the technique of comprehensive coding, which means coding every paragraph in the text. In such cases, it would be useful to implement a function to search for any paragraphs that haven't been coded.

Interesting, I have never heard of such an approach to coding before. Wouldn't you end up drowning in too many codes that way? 

I think you can do this with a search query of the Obsidian core search plugin already. This query would find all paragraphs that have not codes in them `file:{name-of-the-file} line:(". " -"[[")`
